plugin/forward/persistent.go

```diff
@@ -66,85 +64,49 @@ Wait:
 				if time.Since(pc.used) < t.expire {
 					// Found one, remove from pool and return this conn.
 					t.conns[proto] = stack[:len(stack)-1]
-					t.ret <- pc.c
+					t.ret <- pc
 					continue Wait
 				}
-				// clear entire cache if the last conn is expired
-				t.conns[proto] = nil
```

Comment:
so, now the cache is not cleaned until the plugin is stopped? Besides of memory leak this will lead to exhausting open file descriptors (sockets)